lib/src/util/firestore_encoding.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:firedart/firedart.dart';
+import 'package:firedart/src/generated/google/firestore/v1/document.pb.dart'
+    as fs;
+import 'package:firedart/src/generated/google/protobuf/struct.pbenum.dart';
+import 'package:firedart/src/generated/google/protobuf/timestamp.pb.dart';
+import 'package:firedart/src/repository/firestore_gateway.dart';
+import 'package:fixnum/fixnum.dart';
+
+/// An abstract util class which provides firestore decode and encode static methods.
+abstract class FirestoreEncoding {
+  /// Encode dart object into related [fs.Value].
```

Comment:
```suggestion
  /// Encodes Dart object into the corresponding [fs.Value].
```
Do we have to list Dart types supported for encoding? And notice that if `value` has unsupported type then this method throws?